### PR TITLE
Add pgloader bash script to port TCR SQLite DB 

### LINKED ISSUE #2484: Write pgloader script to port SQLite DB to Postgres

### PR DESCRIPTION
**Pull request by m-goggins**
# PULL REQUEST

## Summary
This PR adds a bash script for `pgloader` to load data from the TCR SQLite database into the TEFCA Postgres database. This script should run after the Flyway migrations have been applied to the TEFCADB. Since these migrations are under development, I added some temporary code to ensure that the script is fully executable. 

I know we want to execute the script after the database has been spun up and migrations have been applied so I added it as a separate service in the docker compose file. I'm not sure if this is the best approach but it seemed key to ensure the db has time to spin up before attempting connection. 

## Related Issue
Fixes #2484 

## Acceptance Criteria
Please copy the acceptance criteria from your ticket and paste it here for your reviewer(s)

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Review by robertandremitchell — CHANGES_REQUESTED** [review]
Got this to work by changing permissions with chmod, was running into permission denied before.

```yaml
 entrypoint: ["/bin/sh", "-c", "chmod +x /load_tcr_data.sh && /load_tcr_data.sh"]
```

may just be a me issue, but not sure if it's worth adding that and/or some kind of try logic.

**Review by robertandremitchell — COMMENTED** [review]
When I changed what i think is a typo, I got a new error about `item`, but had trouble replicating it. not sure if it's a real issue, especially if this aspect of the code will change with flyway.


```bash
rob@Roberts-MacBook-Pro tefca-viewer % npm run dev

> tefca-viewer@1.0.1 dev
> docker-compose -f docker-compose-dev.yaml up && next dev

WARN[0000] Found orphan containers ([tefca-viewer-tefca-viewer-1]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up. 
[+] Running 2/1
 ✔ Container tefca-viewer-db-1             C...                               0.1s 
 ✔ Container tefca-viewer-load_tcr_data-1  Created                            0.0s 
Attaching to db-1, load_tcr_data-1
db-1             | The files belonging to this database system will be owned by user "postgres".
db-1             | This user must also own the server process.
db-1             | 
db-1             | The database cluster will be initialized with locale "en_US.utf8".
db-1             | The default database encoding has accordingly been set to "UTF8".
db-1             | The default text search configuration will be set to "english".
db-1             | 
db-1             | Data page checksums are disabled.
db-1             | 
db-1             | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db-1             | creating subdirectories ... ok
db-1             | selecting dynamic shared memory implementation ... posix
db-1             | selecting default max_connections ... 100
db-1             | selecting default shared_buffers ... 128MB
db-1             | selecting default time zone ... UTC
db-1             | creating configuration files ... ok
db-1             | running bootstrap script ... ok
db-1             | sh: locale: not found
db-1             | 2024-09-03 20:55:17.619 UTC [35] WARNING:  no usable system locales were found
db-1             | performing post-bootstrap initialization ... ok
db-1             | syncing data to disk ... ok
db-1             | 
db-1             | initdb: warning: enabling "trust" authentication for local connections
db-1             | initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
db-1             | 
db-1             | Success. You can now start the database server using:
db-1             | 
db-1             |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db-1             | 
db-1             | waiting for server to start....2024-09-03 20:55:18.440 UTC [47] LOG:  starting PostgreSQL 16.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20240309) 13.2.1 20240309, 64-bit
db-1             | 2024-09-03 20:55:18.441 UTC [47] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db-1             | 2024-09-03 20:55:18.449 UTC [50] LOG:  database system was shut down at 2024-09-03 20:55:18 UTC
db-1             | 2024-09-03 20:55:18.468 UTC [47] LOG:  database system is ready to accept connections
db-1             |  done
db-1             | server started
db-1             | CREATE DATABASE
db-1             | 
db-1             | 
db-1             | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
db-1             | 
db-1             | waiting for server to shut down....2024-09-03 20:55:18.700 UTC [47] LOG:  received fast shutdown request
db-1             | 2024-09-03 20:55:18.701 UTC [47] LOG:  aborting any active transactions
db-1             | 2024-09-03 20:55:18.708 UTC [47] LOG:  background worker "logical replication launcher" (PID 53) exited with exit code 1
db-1             | 2024-09-03 20:55:18.708 UTC [48] LOG:  shutting down
db-1             | 2024-09-03 20:55:18.709 UTC [48] LOG:  checkpoint starting: shutdown immediate
db-1             | 2024-09-03 20:55:18.756 UTC [48] LOG:  checkpoint complete: wrote 924 buffers (5.6%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.015 s, sync=0.027 s, total=0.047 s; sync files=301, longest=0.016 s, average=0.001 s; distance=4267 kB, estimate=4267 kB; lsn=0/191AB68, redo lsn=0/191AB68
db-1             | 2024-09-03 20:55:18.766 UTC [47] LOG:  database system is shut down
db-1             |  done
db-1             | server stopped
db-1             | 
db-1             | PostgreSQL init process complete; ready for start up.
db-1             | 
db-1             | 2024-09-03 20:55:18.862 UTC [1] LOG:  starting PostgreSQL 16.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 13.2.1_git20240309) 13.2.1 20240309, 64-bit
db-1             | 2024-09-03 20:55:18.864 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db-1             | 2024-09-03 20:55:18.865 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db-1             | 2024-09-03 20:55:18.866 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db-1             | 2024-09-03 20:55:18.872 UTC [63] LOG:  database system was shut down at 2024-09-03 20:55:18 UTC
db-1             | 2024-09-03 20:55:18.885 UTC [1] LOG:  database system is ready to accept connections
load_tcr_data-1  | CREATE TABLE
load_tcr_data-1  | Table concepts created.
db-1             | 2024-09-03 20:55:21.211 UTC [75] ERROR:  column "item" does not exist at character 44
db-1             | 2024-09-03 20:55:21.211 UTC [75] STATEMENT:  INSERT INTO concepts (id, name) VALUES (1, item);
load_tcr_data-1  | ERROR:  column "item" does not exist
load_tcr_data-1  | LINE 1: INSERT INTO concepts (id, name) VALUES (1, item);
load_tcr_data-1  |                                                    ^
load_tcr_data-1  | CREATE TABLE
load_tcr_data-1  | Table valuesets created.
db-1             | 2024-09-03 20:55:21.387 UTC [77] ERROR:  column "item" does not exist at character 45
db-1             | 2024-09-03 20:55:21.387 UTC [77] STATEMENT:  INSERT INTO valuesets (id, name) VALUES (1, item);
load_tcr_data-1  | ERROR:  column "item" does not exist
load_tcr_data-1  | LINE 1: INSERT INTO valuesets (id, name) VALUES (1, item);
load_tcr_data-1  |                                                     ^
load_tcr_data-1  | CREATE TABLE
load_tcr_data-1  | Table conditions created.
db-1             | 2024-09-03 20:55:21.566 UTC [79] ERROR:  column "item" does not exist at character 46
db-1             | 2024-09-03 20:55:21.566 UTC [79] STATEMENT:  INSERT INTO conditions (id, name) VALUES (1, item);
load_tcr_data-1  | ERROR:  column "item" does not exist
load_tcr_data-1  | LINE 1: INSERT INTO conditions (id, name) VALUES (1, item);
load_tcr_data-1  |                                                      ^
load_tcr_data-1  | Tables are empty. Proceeding with data loading...
load_tcr_data-1  | Downloading ERSD SQLite file...
load_tcr_data-1  | Connecting to github.com (140.82.112.4:443)
load_tcr_data-1  | Connecting to raw.githubusercontent.com (185.199.111.133:443)
load_tcr_data-1  | saving to '/tmp/ersd.db'
load_tcr_data-1  | ersd.db              100% |********************************| 29.3M  0:00:00 ETA
load_tcr_data-1  | '/tmp/ersd.db' saved
load_tcr_data-1  | Loading data into PostgreSQL...
load_tcr_data-1  | 2024-09-03T20:55:25.079000Z LOG pgloader version "3.6.9"
load_tcr_data-1  | 2024-09-03T20:55:25.358999Z LOG Migrating from #<SQLITE-CONNECTION sqlite:///tmp/ersd.db {1005A71F03}>
load_tcr_data-1  | 2024-09-03T20:55:25.358999Z LOG Migrating into #<PGSQL-CONNECTION pgsql://postgres@db:5432/tefca_db {1005B7E8A3}>
db-1             | 2024-09-03 20:55:28.303 UTC [128] ERROR:  multiple primary keys for table "valuesets" are not allowed
db-1             | 2024-09-03 20:55:28.303 UTC [128] STATEMENT:  ALTER TABLE valuesets ADD PRIMARY KEY USING INDEX idx_16399_sqlite_autoindex_valuesets_1;
load_tcr_data-1  | 2024-09-03T20:55:28.029000Z ERROR PostgreSQL Database error 42P16: multiple primary keys for table "valuesets" are not allowed
load_tcr_data-1  | QUERY: ALTER TABLE valuesets ADD PRIMARY KEY USING INDEX idx_16399_sqlite_autoindex_valuesets_1;
db-1             | 2024-09-03 20:55:28.313 UTC [128] ERROR:  multiple primary keys for table "conditions" are not allowed
db-1             | 2024-09-03 20:55:28.313 UTC [128] STATEMENT:  ALTER TABLE conditions ADD PRIMARY KEY USING INDEX idx_16404_sqlite_autoindex_conditions_1;
load_tcr_data-1  | 2024-09-03T20:55:28.037000Z ERROR PostgreSQL Database error 42P16: multiple primary keys for table "conditions" are not allowed
load_tcr_data-1  | QUERY: ALTER TABLE conditions ADD PRIMARY KEY USING INDEX idx_16404_sqlite_autoindex_conditions_1;
db-1             | 2024-09-03 20:55:28.319 UTC [128] ERROR:  multiple primary keys for table "concepts" are not allowed
db-1             | 2024-09-03 20:55:28.319 UTC [128] STATEMENT:  ALTER TABLE concepts ADD PRIMARY KEY USING INDEX idx_16409_sqlite_autoindex_concepts_1;
load_tcr_data-1  | 2024-09-03T20:55:28.043000Z ERROR PostgreSQL Database error 42P16: multiple primary keys for table "concepts" are not allowed
load_tcr_data-1  | QUERY: ALTER TABLE concepts ADD PRIMARY KEY USING INDEX idx_16409_sqlite_autoindex_concepts_1;
db-1             | 2024-09-03 20:55:28.332 UTC [128] ERROR:  there is no unique constraint matching given keys for referenced table "concepts"
db-1             | 2024-09-03 20:55:28.332 UTC [128] STATEMENT:  ALTER TABLE crosswalk ADD FOREIGN KEY(icd10_code) REFERENCES concepts(gem_formatted_code) ON UPDATE NO ACTION ON DELETE NO ACTION
load_tcr_data-1  | 2024-09-03T20:55:28.056000Z ERROR PostgreSQL Database error 42830: there is no unique constraint matching given keys for referenced table "concepts"
load_tcr_data-1  | QUERY: ALTER TABLE crosswalk ADD FOREIGN KEY(icd10_code) REFERENCES concepts(gem_formatted_code) ON UPDATE NO ACTION ON DELETE NO ACTION
db-1             | 2024-09-03 20:55:28.409 UTC [128] ERROR:  there is no unique constraint matching given keys for referenced table "concepts"
db-1             | 2024-09-03 20:55:28.409 UTC [128] STATEMENT:  ALTER TABLE icd_crosswalk ADD FOREIGN KEY(icd10_code) REFERENCES concepts(gem_formatted_code) ON UPDATE NO ACTION ON DELETE NO ACTION
load_tcr_data-1  | 2024-09-03T20:55:28.132999Z ERROR PostgreSQL Database error 42830: there is no unique constraint matching given keys for referenced table "concepts"
load_tcr_data-1  | QUERY: ALTER TABLE icd_crosswalk ADD FOREIGN KEY(icd10_code) REFERENCES concepts(gem_formatted_code) ON UPDATE NO ACTION ON DELETE NO ACTION
load_tcr_data-1  | 2024-09-03T20:55:28.144999Z LOG report summary reset
load_tcr_data-1  |              table name     errors       rows      bytes      total time
load_tcr_data-1  | -----------------------  ---------  ---------  ---------  --------------
load_tcr_data-1  |                   fetch          0          0                     0.000s
load_tcr_data-1  |         fetch meta data          0         27                     0.152s
load_tcr_data-1  |          Create Schemas          0          0                     0.001s
load_tcr_data-1  |        Create SQL Types          0          0                     0.015s
load_tcr_data-1  |           Create tables          0         14                     0.077s
load_tcr_data-1  |          Set Table OIDs          0          7                     0.011s
load_tcr_data-1  | -----------------------  ---------  ---------  ---------  --------------
load_tcr_data-1  |               crosswalk          0      81593     1.9 MB          0.629s
load_tcr_data-1  |               valuesets          0       1399   216.3 kB          0.204s
load_tcr_data-1  |                concepts          0      45273     6.3 MB          1.068s
load_tcr_data-1  |              conditions          0        226    17.6 kB          0.491s
load_tcr_data-1  |   condition_to_valueset          0       1513    99.6 kB          0.519s
load_tcr_data-1  |           icd_crosswalk          0      81593     1.9 MB          0.878s
load_tcr_data-1  |     valueset_to_concept          0      45273     3.8 MB          0.854s
load_tcr_data-1  | -----------------------  ---------  ---------  ---------  --------------
load_tcr_data-1  | COPY Threads Completion          0          4                     1.380s
load_tcr_data-1  |          Create Indexes          0         14                     1.297s
load_tcr_data-1  |  Index Build Completion          0         14                     0.477s
load_tcr_data-1  |         Reset Sequences          0          0                     0.140s
load_tcr_data-1  |            Primary Keys          3          7                     0.050s
load_tcr_data-1  |     Create Foreign Keys          2          4                     0.079s
load_tcr_data-1  |         Create Triggers          0          0                     0.000s
load_tcr_data-1  |        Install Comments          0          0                     0.000s
load_tcr_data-1  | -----------------------  ---------  ---------  ---------  --------------
load_tcr_data-1  |       Total import time          ✓     256870    14.3 MB          3.423s
load_tcr_data-1  | Successfully loaded TCR data into tefca_db
load_tcr_data-1 exited with code 0
```

File: containers/tefca-viewer/load_tcr_data.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Define the database connection
+PG_CONN="postgresql://postgres:pw@db:5432/tefca_db"
+
+tables=("concepts" "valuesets" "conditions")
+
+### TEMP: Remove this once flyway is implemented
+for table in "${tables[@]}"; do
+    psql "$PG_CONN" -c "CREATE TABLE IF NOT EXISTS $table (id INT, name VARCHAR(80));" -tA
+    echo "Table $table created."
+    pgsl "$PG_CONN" -c "INSERT INTO $table (id, name) VALUES (1, item);"
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply containers/tefca-viewer/load_tcr_data.sh:14]: Def should be `psql`, good catch!

[the remark this answers is not in the document]
Comment (m-goggins) [reply containers/tefca-viewer/load_tcr_data.sh:14]: the issues with `item` should go away once Brandon's flyway work is merged. We could wait to have his pr merged before retrying.

[the remark this answers is not in the document]
Comment (robertandremitchell) [reply containers/tefca-viewer/load_tcr_data.sh]: makes sense!

**Review by robertandremitchell — APPROVED** [review]

**Pull request merged**